core/order.go

```diff
@@ -1183,6 +1219,10 @@ collectListings:
 	if err != nil {
 		return err
 	}
+
+	if !knownListings {
+		return UnknowListingError
+	}
```

Comment:
And maybe since the only usage so far is a negative, maybe it should be `n.HasNoListings()` or maybe what this REALLY depends on in the domain and name it `n.IsVendor()` but I don't know if that's too broad for this case. This is how we can capture domain knowledge without having to pass it around by word-of-mouth or trying to keep up with out-of-date docs.